Composer/packages/client/src/recoilModel/dispatchers/dialogs.ts

```diff
@@ -24,7 +25,16 @@ import { removeDialogSchema } from './dialogSchema';
 export const dialogsDispatcher = () => {
   const removeDialog = useRecoilCallback(
     (callbackHelpers: CallbackInterface) => async (id: string, projectId: string) => {
-      const { set, reset } = callbackHelpers;
+      const { set, reset, snapshot } = callbackHelpers;
+
+      const dialog = await snapshot.getPromise(dialogState({ projectId, dialogId: id }));
+
+      // If the dialog is a generated form dialog, delete using form dialog dispatcher
+      if (dialog.content?.$schema) {
```

Comment:
Do you think we should have a shared helper function for isDialogFormDialog?

Reply:
probably yes.